diff --git a/minui/graphics_fbdev.cpp b/minui/graphics_fbdev.cpp
index 5f64e1e..f78f353 100644
--- a/minui/graphics_fbdev.cpp
+++ b/minui/graphics_fbdev.cpp
@@ -54,7 +54,11 @@ static void fbdev_blank(minui_backend* backend __unused, bool blank)
 #else
     int ret;
 
+#ifdef TW_SCREEN_BLANK_ON_BOOT
+    ret = ioctl(fb_fd, FBIOBLANK, FB_BLANK_UNBLANK);
+#else
     ret = ioctl(fb_fd, FBIOBLANK, blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK);
+#endif
     if (ret < 0)
         perror("ioctl(): blank");
 #endif
diff --git a/minuitwrp/graphics_fbdev.cpp b/minuitwrp/graphics_fbdev.cpp
index a1c42d0..4132b9a 100644
--- a/minuitwrp/graphics_fbdev.cpp
+++ b/minuitwrp/graphics_fbdev.cpp
@@ -87,7 +87,11 @@ static void fbdev_blank(minui_backend* backend __unused, bool blank)
 #ifndef TW_NO_SCREEN_BLANK
     int ret;
 
+#ifdef TW_SCREEN_BLANK_ON_BOOT
+    ret = ioctl(fb_fd, FBIOBLANK, FB_BLANK_UNBLANK);
+#else
     ret = ioctl(fb_fd, FBIOBLANK, blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK);
+#endif
     if (ret < 0)
         perror("ioctl(): blank");
 #endif
diff --git a/minuitwrp/graphics_overlay.cpp b/minuitwrp/graphics_overlay.cpp
index 84ea6e0..26b1228 100644
--- a/minuitwrp/graphics_overlay.cpp
+++ b/minuitwrp/graphics_overlay.cpp
@@ -175,7 +175,11 @@ static void overlay_blank(minui_backend* backend __unused, bool blank)
 #else
     int ret;
 
+#ifdef TW_SCREEN_BLANK_ON_BOOT
+    ret = ioctl(fb_fd, FBIOBLANK, FB_BLANK_UNBLANK);
+#else
     ret = ioctl(fb_fd, FBIOBLANK, blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK);
+#endif
     if (ret < 0)
         perror("ioctl(): blank");
 #endif
